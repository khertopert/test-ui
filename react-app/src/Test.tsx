import { useState } from "react";

export function Test() {
    const [hidden, setHidden] = useState(false);

    return <>
        {!hidden && <div data-testid="description-text">
            <span>Hi, i am Test component</span>
        </div>}
        <button data-testid="hidden-button" onClick={() => setHidden(!hidden)}>hidden</button>
    </>
}