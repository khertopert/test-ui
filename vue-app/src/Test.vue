<script setup lang="ts">
import {ref} from "vue";

const hidden = ref(false);

</script>

<template>
    <div v-if="!hidden">
        <span>Hi, i am Test component</span>
    </div>
    <button @click="() => hidden = !hidden">hidden</button>
</template>