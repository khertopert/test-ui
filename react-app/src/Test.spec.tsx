import { render, fireEvent } from '@testing-library/react';
import { Test } from "./Test";

it('should render component', () => {
    const { asFragment } = render(<Test></Test>);
    expect(asFragment()).toMatchSnapshot();
});

it('should render component (v2)', () => {
    const { getByTestId } = render(<Test></Test>);

    const descriptionText = getByTestId('description-text') as HTMLDivElement;
    expect(descriptionText).not.toBeNull();
});

it('should hide description when button is clicked', () => {
    const { asFragment, getByTestId } = render(<Test></Test>);

    const button = getByTestId('hidden-button') as HTMLButtonElement;
    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
});

it('should hide description when button is clicked (v2)', () => {
    const { queryByTestId, getByTestId } = render(<Test></Test>);

    const button = getByTestId('hidden-button') as HTMLButtonElement;
    fireEvent.click(button);

    const descriptionText = queryByTestId('description-text') as HTMLDivElement;
    expect(descriptionText).toBeNull();
});