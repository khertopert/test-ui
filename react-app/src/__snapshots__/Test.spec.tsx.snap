// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide description when button is clicked 1`] = `
<DocumentFragment>
  <button
    data-testid="hidden-button"
  >
    hidden
  </button>
</DocumentFragment>
`;

exports[`should render component 1`] = `
<DocumentFragment>
  <div
    data-testid="description-text"
  >
    <span>
      Hi, i am Test component
    </span>
  </div>
  <button
    data-testid="hidden-button"
  >
    hidden
  </button>
</DocumentFragment>
`;
